src/common/expansion.rs

```diff
@@ -0,0 +1,54 @@
+use regex::Regex;
+use std::{collections::HashMap, sync::LazyLock};
+
+use crate::event::Value;
+
+static RE: LazyLock<Regex> = LazyLock::new(|| Regex::new(r"[^0-9A-Za-z_]").unwrap());
+fn slugify_text(input: String) -> String {
+    let result = RE.replace_all(&input, "_");
+    result.to_lowercase()
+}
+
+pub(crate) fn pair_expansion(
+    key_s: String,
+    value_s: String,
```

Comment:
```suggestion
    key_s: &str,
    value_s: &str,
```

This will require some usage updates.